### PR TITLE
Support importing of V2 docs with formulas

### PR DESCRIPTION
**Pull request by pjanik**
This PR adds support for importing V2 docs with formulas.
It required a couple of fixes, but the most important one was to change the formula's source of truth to display string.
The canonical variant is computed as a derived value, so in most cases, it should be cached (that's why I've added `observer` to a few components). I bet over time, we might decide that it's better to store canonical string too, but for now, it simplifies quite a few things (no need to worry about synchronization between these two strings).

### CODE REVIEW

File: v3/src/models/data/formula.ts [hunk 1]
```diff
@@ -5,29 +5,29 @@ import { getFormulaManager } from "../tiles/tile-environment"
 
 export const Formula = types.model("Formula", {
   id: types.optional(types.identifier, () => typedId("FORMULA")),
```
Comment (kswenson) [inline v3/src/models/data/formula.ts:7]: Conventionally, the `typedId` prefix is four characters, so this would be `FORM`.

**Review by kswenson — APPROVED** [review]
👍 Looks good

File: v3/src/components/case-table/attribute-menu/attribute-menu-list.tsx [hunk 2]
```diff
@@ -1,19 +1,21 @@
 import React, { forwardRef } from "react"
+
 import { MenuItem, MenuList, useDisclosure, useToast } from "@chakra-ui/react"
 import { useCaseMetadata } from "../../../hooks/use-case-metadata"
 import { useDataSetContext } from "../../../hooks/use-data-set-context"
 import { TCalculatedColumn } from "../case-table-types"
 import { EditAttributePropertiesModal } from "./edit-attribute-properties-modal"
 import t from "../../../utilities/translation/translate"
 import { EditFormulaModal } from "./edit-formula-modal"
+import { observer } from "mobx-react-lite"
```
[the remark this answers is not in the document]
Comment (pjanik) [reply v3/src/components/case-table/attribute-menu/attribute-menu-list.tsx:10]: Doh, forgotten VSCode auto-import again. I need to know what was the result of @scytacki's spike 😉 There are two ESlint packages that can order imports for us automatically, and they have lots of rules - I'd hope we could replicate our patterns.

File: v3/src/models/data/formula.ts (hunk 1 above)
Comment (pjanik) [reply v3/src/models/data/formula.ts]: Thanks, fixed.

**Pull request merged**